### PR TITLE
Feature/265 data officer role

### PR DESCRIPTION
**Pull request by Ruby77**
## Description

Created the new data officer role and created CRUD functionality for institutions.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

1. Check the wiki and check the permissions page. This page should be upaded and should contain the new data officer role and the new endoints for the institution controller. Also, the existing endpoints that can be reachable for data officer should be updated.
2. Run the postman tests and check that no test fails (Response time is less than ....ms tests excluded).
3. Run the InstitutionServiceTests and check that no test fails.
4. Run the InstitutionRepositoryTests and check that no test fails.

## Link to issue

Closes: #265

### CODE REVIEW

**Review by DaveBouman — APPROVED** [review]

**Review by MeesvanStraten — APPROVED** [review]
Looks good! No real comments. Postman tests are all passing :).

**Review by Brend-Smits — CHANGES_REQUESTED** [review]
I will be reviewing this today, don't merge yet.

**Review by niraymak — CHANGES_REQUESTED** [review]
Nice solid code! 🥇 

Will approve after very little requests mentioned in Teams conversation!

**Review by niraymak — APPROVED** [review]

File: IdentityServer/Startup.cs
Comment (Brend-Smits) [inline IdentityServer/Startup.cs:235]: Something probably went wrong here while merging. Please check the file on develop: https://github.com/DigitalExcellence/dex-backend/blob/develop/IdentityServer/Startup.cs#L233-L252

Comment (Brend-Smits) [reply IdentityServer/Startup.cs:235]: Make sure the seed file is also properly merged with develop

**Review by Brend-Smits — CHANGES_REQUESTED** [review]
Looks great, nice work. Please check my comment and make sure the merge from develop into this branch succeeded.

**Review by Brend-Smits — APPROVED** [review]

**Review by niraymak — APPROVED** [review]

**Review by DaveBouman — APPROVED** [review]

**Pull request merged**
